modules/programs/mpv.nix

```
@@ -0,0 +1,142 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (builtins) typeOf stringLength;
+
+  cfg = config.programs.mpv;
+
+  mpvOption = with types; either str (either int (either bool float));
+  mpvOptions = with types; attrsOf mpvOption;
+  mpvProfiles = with types; attrsOf mpvOptions;
+  mpvBindings = with types; attrsOf str;
+
+  renderOption = option:
+    rec {
+      int = toString option;
+      float = int;
+
+      bool = if option then "yes" else "no";
+
+      string = option;
+
+      null = "";
+    }.${typeOf option};
+
+  renderOptions = options:
+    concatStringsSep "\n"
+      (mapAttrsToList
+        (name: value:
+          let
+            rendered = renderOption value;
+            length = toString (stringLength rendered);
+          in
+          "${name}=%${length}%${rendered}")
+          options);
+
+  renderProfiles = profiles:
+    concatStringsSep "\n"
+      (mapAttrsToList
+        (name: value: ''
+          [${name}]
+          ${renderOptions value}
+        '')
+        profiles);
+
+  renderBindings = bindings:
+    concatStringsSep "\n"
+      (mapAttrsToList
+        (name: value:
+          "${name} ${value}")
+        bindings);
+
+in {
+  options = {
+    programs.mpv = {
+      enable = mkEnableOption "mpv";
+
+      config = mkOption {
+        description = ''
+          Configuration written to ~/.config/mpv/mpv.conf.
```

Can make a `<filename>` element:
```suggestion
          Configuration written to <filename>~/.config/mpv/mpv.conf</filename>.
```